src/Guild.js

```diff
@@ -25,6 +25,35 @@ class Guild {
     autoBind(this);
   }
 
+  guildRoleCreate (roleId) {
+    this.roles[this.roles.name] = roleId;
+  }
+
+  guildGetKeyByValue (value) {
+    return Object.keys(this.roles).find(key => this.roles[key] === value);
+  }
+
+  guildUpdateRoleName (roleId) {
+    const oldKey = this.guildGetKeyByValue(roles.id);
+
+    //role for roleId does not exist. nothing to update
+    if(!this.isRole(oldKey)) {
+      return;
+    }
+
+    if(oldKey !== this.roles.name) {
+      this.roles[this.roles.name] = this.roles[oldKey];
```

Comment:
You already have the id of the the Role. You don't need to get it again from `this.roles[oldKey]`.

Reply:
So, here i am assigning the old keys value to the new key before deleting the old key/value pair.
if i do nothing here and deleting the old key/value pair. the new key wont have a value.

Not sure if i get what you mean here.

Reply:
If the Role name has changed, the ID will be the same. Therefore you can just use the passed ID and assign that to the new key. Does that make more sense?